### ISSUE TITLE
Docker init - Custom template support

### ISSUE THREAD

**Issue by minwook-shin** [docker/roadmap#483]
**Tell us about your request**
I hope support the custom template of "docker init" command.

issue : https://github.com/docker/cli/issues/4184

**Which service(s) is this request for?**
Docker CLI

how about get the URL of Github repository similar to cookie cutter template (ex: psql to Maria DB, or built-in lib case)?
ex) real cookiecutter sample code

```
{
  "project_name": "Project Name",
  "version": "0.1",
  "use_postgres": "yes",
}
```

```
{%- set postgres = cookiecutter.use_postgres in ('y', 'yes') -%}
```
and how about supporting a good template to upload Extensions Marketplace?
ex) This is a sample URL, does not exist.
```
https://hub.docker.com/search?q=&type=template
```

**Are you currently working around the issue?**
No, it cant be resolved because it is private source code.

**Comment by nunocoracao** [comment]
@Neirda24 @minwook-shin thanks for your feedback we are actively considering this and I would be interested in gathering more information on how you would leverage this feature at your workplaces

**Comment by conradwt** [comment]
Yes, I would definitely like to be able to customize the existing templates as well create new templates for unsupported languages without having to submit a PR.

**Comment by tylertitsworth** [comment]
I have a use case to create templates related to AI Frameworks (TensorFlow, PyTorch, Classical ML, etc.). Docker init is a great idea that could be significantly expanded upon by Open Source Contributions.

For instance, being able to identify os dependencies from certain python packages using a RAG LLM.
